src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/StandardCallerArgumentCollection.java

```diff
@@ -72,37 +72,75 @@ public void copyStandardCallerArgsFrom( final StandardCallerArgumentCollection o
     @Argument(fullName = "contamination-fraction-per-sample-file", shortName = "contamination-file", doc = "Tab-separated File containing fraction of contamination in sequencing data (per sample) to aggressively remove. Format should be \"<SampleID><TAB><Contamination>\" (Contamination is double) per line; No header.", optional = true)
     public File CONTAMINATION_FRACTION_FILE = null;
 
+    private DefaultedMap<String,Double> sampleContamination;
+    private boolean mapHasContaminationSet = false;
+
     /**
      * Returns true if there is some sample contamination present, false otherwise.
      * @return {@code true} iff there is some sample contamination
      */
     public boolean isSampleContaminationPresent() {
-        return (!Double.isNaN(CONTAMINATION_FRACTION) && CONTAMINATION_FRACTION > 0.0) || (sampleContamination != null && !sampleContamination.isEmpty());
+        return contaminationFractionIsSet(CONTAMINATION_FRACTION) || mapHasContaminationSet;
     }
-
-    private DefaultedMap<String,Double> sampleContamination;
-
+    
     /**
      * Returns an unmodifiable view of the map of SampleId -> contamination.
+     * 
+     * The returned map will return a default value equal to the configured
+     * {@link #CONTAMINATION_FRACTION} for samples whose contamination is not
+     * explicitly set.
      */
     public Map<String,Double> getSampleContamination() {
+        if (sampleContamination == null) {
+            setSampleContamination(Collections.emptyMap()); // default to empty map
+        }
         return Collections.unmodifiableMap(sampleContamination);
     }
 
     /**
-     * Returns the sample contamination or CONTAMINATION_FRACTION if no contamination level was specified for this sample.
+     * Set the sample contamination map using the provided map. The resulting map will have
+     * its default value for unknown keys set equal to {@link #CONTAMINATION_FRACTION}, regardless
+     * of any default value set in the provided map (if it's a DefaultedMap).
+     *
+     * @param sampleContamination Map of sample to contamination fraction with which to initialize our
+     *                            sample contamination map. Replaces any existing values in our map.
+     *                            The resulting map will have {@link #CONTAMINATION_FRACTION} as the default
+     *                            value for unknown keys, regardless of any default set in the provided map.
      */
-    public Double getSampleContamination(final String sampleId){
-        Utils.nonNull(sampleId);
-        if (sampleContamination == null){
-            setSampleContamination(new DefaultedMap<>(CONTAMINATION_FRACTION));//default to empty map
-        }
-        return sampleContamination.get(sampleId);
-    }
-
-    public void setSampleContamination(final DefaultedMap<String, Double> sampleContamination) {
+    public void setSampleContamination(final Map<String, Double> sampleContamination) {
```

Comment:
Why does this code belong to the argument collection? I understand it is needs access to the argument collection values in order to fill defaults, but this code is not a user argument, indeed it needs to be called into by the tool itself in order to poke contamination back into its own argument collection? Could this logic live elsewhere?

Reply:
Well, there is precedent in the GATK for arg collections containing code to parse/load their arguments (see `IntervalArgumentCollection` for an example). The design of this particular class, however, in which the contamination file must be loaded externally and passed in via a setter, is not great. I'll create a ticket to refactor this in a future PR, but I'd like to get the actual fix out first.